cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -359,6 +359,31 @@ pub fn start_service(socket: impl AsRef<Path>, name: impl AsRef<str>) -> Result<
     }
 }
 
+/// Restart service using `process-compose process restart`.
+pub fn restart_service(
+    socket: impl AsRef<Path>,
+    name: impl AsRef<str>,
+) -> Result<(), ServiceError> {
+    tracing::debug!(name = name.as_ref().to_string(), "restarting service");
+
+    let mut cmd = base_process_compose_command(socket);
+    let output = cmd
+        .args(["restart", name.as_ref()])
+        .stderr(Stdio::piped())
+        .stdout(Stdio::piped())
+        .output()
```

Comment:
non-blocking:

```suggestion
        .output()
```

`Command::output()` will set the outputs to Piped by default
Fine if we include in the follow up.

Reply:
I've included it in https://github.com/flox/flox/pull/1923 plus the other places we do it 👍🏻